File: src/main/java/net/hectus/neobb/turn/legacy_game/warp/LTUnderwaterWarp.java
package net.hectus.neobb.turn.legacy_game.warp;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.default_game.attributes.clazz.*;
import org.bukkit.World;

import java.util.List;

public class LTUnderwaterWarp extends LWarpTurn {
    public LTUnderwaterWarp(World world) { super(world, "legacy-underwater"); }
    public LTUnderwaterWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "legacy-underwater"); }

    @Override
    public void apply() {
        super.apply();
        new Buff.Luck(Buff.BuffTarget.YOU, 8);
    }

    @Override
    public int chance() {
        return 73;
    }

    @Override
    public List<Class<? extends Clazz>> allows() {
        return List.of(NeutralClazz.class, ColdClazz.class, WaterClazz.class, NatureClazz.class);
    }

    @Override
    public int cost() {
        return 4;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/legacy_game/warp/LTUnderwaterWarp.kt
package net.hectus.neobb.modes.turn.legacy_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.matrix.structure.PlacedStructure
import net.hectus.neobb.matrix.structure.StaticStructure
import net.hectus.neobb.matrix.structure.StaticStructures
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.*
import net.hectus.neobb.player.NeoPlayer
import kotlin.reflect.KClass

class LTUnderwaterWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : LWarpTurn(data, cord, "underwater", player) {
    override val staticStructure: StaticStructure = StaticStructures.Legacy.Warp.UNDERWATER

    override val cost: Int = 4
    override val chance: Double = 73.0
    override val allows: List<KClass<out Clazz>> = listOf(NeutralClazz::class, ColdClazz::class, WaterClazz::class, NatureClazz::class)
    override val temperature: Temperature = Temperature.NORMAL

    override fun apply() {
        Luck(8).apply(player!!)
    }
}
